Main.py

```diff
@@ -117,6 +117,15 @@ def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = No
         for item_name, count in world.start_inventory_from_pool.setdefault(player, StartInventoryPool({})).value.items():
             for _ in range(count):
                 world.push_precollected(world.create_item(item_name, player))
+            # remove from_pool items also from early items handling, as starting is plenty early.
```

Comment:
I think this snippet would be way easier to read and understand with slightly improved variable names. For example, "early" is used twice, in mutually exclusive usages (assign, then read, then assign again, then read again), so if it was two variables, making the difference between both usages more clear, it would be a good improvement